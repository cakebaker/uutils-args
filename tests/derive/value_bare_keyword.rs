use uutils_args::{Arguments, Options, Value};

#[derive(Value, Default)]
enum Flavor {
    #[default]
    #[value(banana)] // Oops!
    Banana,
}

#[derive(Arguments)]
enum Arg {
    #[arg("--flavor=FLAVOR")]
    Flavor(Flavor),
}

struct Settings {}

impl Options<Arg> for Settings {
    fn apply(&mut self, _arg: Arg) -> Result<(), uutils_args::Error> {
        Ok(())
    }
}

fn main() {
    Settings {}.parse(std::env::args_os()).unwrap();
}
