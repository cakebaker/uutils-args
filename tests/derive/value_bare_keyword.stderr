error: proc-macro derive panicked
 --> tests/derive/value_bare_keyword.rs:3:10
  |
3 | #[derive(Value, Default)]
  |          ^^^^^
  |
  = help: message: expected comma-separated list of string literals: Error("unexpected end of input, unrecognized keyword in value attribute")

error[E0277]: the trait bound `Flavor: uutils_args::Value` is not satisfied
  --> tests/derive/value_bare_keyword.rs:10:10
   |
10 | #[derive(Arguments)]
   |          ^^^^^^^^^ the trait `uutils_args::Value` is not implemented for `Flavor`
   |
   = help: the following other types implement trait `uutils_args::Value`:
             Option<T>
             OsString
             PathBuf
             String
             i128
             i16
             i32
             i64
           and $N others
note: required by a bound in `parse_value_for_option`
  --> src/internal.rs
   |
   | pub fn parse_value_for_option<T: Value>(opt: &str, v: &OsStr) -> Result<T, ErrorKind> {
   |                                  ^^^^^ required by this bound in `parse_value_for_option`
   = note: this error originates in the derive macro `Arguments` (in Nightly builds, run with -Z macro-backtrace for more info)
